### DISCUSSION TITLE
rustc 1.60 is required

### DISCUSSION THREAD

**Discussion by webern** [kube-rs/kube#921] [General]
Is there a reason rustc >= 1.60 is required?

```
package `kube v0.73.0` cannot be built because it requires rustc 1.60.0 or newer, while the currently active rustc version is 1.58.1
```

**Comment by clux** [comment]
yeah:

- new k8s-openapi pulled in 1.58 with [f strings](https://www.rustnote.com/blog/format_strings.html)
- we skipped some versions due to https://github.com/kube-rs/kube-rs/issues/746

**Comment by shanesveller** [comment]
FWIW this was [announced in the changelog](https://github.com/kube-rs/kube-rs/blob/80272c04ae0036983bb0de4d522003cfd1f5c189/CHANGELOG.md#0730--2022-05-23) but is not highlighted with a README entry if you're a brand new user.

**Reply by webern** [reply]
Thank you, sorry I missed that.

**Comment by clux** [comment]
There is also a small badge for it in the README;
> [![Rust 1.60](https://img.shields.io/badge/MSRV-1.60-dea584.svg)](https://github.com/rust-lang/rust/releases/tag/1.60.0)

...btw is 1.60 problematic?

we [considered putting a 3 month hard policy](https://github.com/kube-rs/kube-rs/issues/668) in the past, but we currently just go with latest stable if there's stuff in there we want.

**Reply by webern** [reply]
Projects that require reproducible builds don't necessarily update their compiler right away. In our case, I think our SDK is currently in the process of being bumped to 1.60. In general, it's probably better for the ecosystem to support at least a few minor versions of the compiler rather than requiring the absolute latest compiler.

**Reply by clux** [reply]
Yeah, that is reasonable. We already had "we try to let the MSRV trail one or two stable versions behind the latest stable" under https://kube.rs/rust-version/ , but can open a PR to see if we can strengthen that to `2` minor versions (where feasible).
EDIT: https://github.com/kube-rs/website/pull/17
